### PR TITLE
Fixes #2024 Add bulk export commands to CLI

### LINKED ISSUE #2024: Add "bulk" commands to the Medplum CLI
Add "bulk" commands to the Medplum CLI

Use case is to pull data from  https://bcda.cms.gov/guide.html#try-the-api and load it into a Medplum project.

medplum bulk export x where x is a Group/id, a Patient/id or a system level export
medplum bulk import filename.ndjson to import into your current project

You should be able to do this with just the medplum CLI then:

```
# Use BCDA client credentials
export MEDPLUM_CLIENT_ID=$BCDA_CLIENT_ID
export MEDPLUM_CLIENT_SECRET=$BCDA_CLIENT_SECRET
export MEDPLUM_BASE_URL=$BCDA_BASE_URL
medplum bulk export Patient > myfile.ndjson

# Use Medplum client credentials
export MEDPLUM_CLIENT_ID=$MY_CLIENT_ID
export MEDPLUM_CLIENT_SECRET=$MY_CLIENT_SECRET
export MEDPLUM_BASE_URL=https://api.medplum.com/
medplum bulk import myfile.ndjson

# Or use Medplum user login
medplum login
medplum bulk import myfile.ndjson
```

### PR DESCRIPTION
**Pull request by tonylee80**
depends on #2071,
In addition, bulk import will be implemented in another PR

```
tonylee@tony-mbp medplum % npx medplum bulk export --help
Usage: medplum bulk-export [options]

Options:
  -e, --exportLevel <exportLevel>  Optional export level. Defaults to system level export. "Group/:id" - Group of Patients, "Patient" - All Patients.
  -t, --types <types>              optional resource types to export
  -s, --since <since>              optional Resources will be included in the response if their state has changed after the supplied time (e.g. if Resource.meta.lastUpdated is later than the
                                   supplied _since time).
  -h, --help                       display help for command
  ```

```
tonylee@tony-mbp test % npx medplum bulk export --types "ProjectMembership,Group,Project"
Group.json is created
Project.json is created
ProjectMembership.json is created

tonylee@tony-mbp test % cat Group.json 
{"resourceType":"Group","type":"person","active":false,"actual":true,"id":"683652af-76f5-4f4a-9b1f-7c6490a502fc","meta":{"versionId":"f68fd7b0-7dc8-4afc-8863-8c99f5cb0876","lastUpdated":"2023-05-12T22:14:21.787Z","author":{"reference":"Practitioner/f0fb606e-2c6a-4ba7-a170-cfb6ebe2926a","display":"Tony Lee"},"project":"8aee2f1e-6278-44a6-b9b1-f37fac33b14d","compartment":[{"reference":"Project/8aee2f1e-6278-44a6-b9b1-f37fac33b14d"},{"reference":"Patient/823d06d0-b608-4acd-a131-77ade43c4ec0","display":"Test Test"}]},"member":[{"entity":{"reference":"Patient/823d06d0-b608-4acd-a131-77ade43c4ec0","display":"Test Test"}},{"entity":{"reference":"Patient/584fd8e9-9096-4947-9385-e7a513b5e027","display":"Bob965 Bins636"}},{"entity":{"reference":"Patient/2115722a-4802-4937-9d3e-f239ae1ceabe","display":"Ms. Ta701 Zieme486"}}]}
{"resourceType":"Group","type":"person","id":"f0a34596-91d7-43f7-94f1-9763acf6a053","meta":{"versionId":"b91618c9-f057-45f3-9f1c-9a72f89f47a8","lastUpdated":"2023-05-21T10:05:18.225Z","author":{"reference":"Practitioner/93ea230d-323c-410f-aeb7-98b6e96b21ab","display":"Medplum Admin"},"project":"3a6968bf-91d4-4140-a572-97f979857270","compartment":[{"reference":"Project/3a6968bf-91d4-4140-a572-97f979857270"}]},"managingEntity":{"reference":"Practitioner/6d76ee5e-af29-45e4-b21e-89c2d0f1c18f","display":"Dr. Teodoro374 Jaskolski867"}}
tonylee@tony-mbp test % cat Project.json
{"resourceType":"Project","name":"Canada Sample Data","owner":{"reference":"User/c3c2eb49-0317-438e-b42c-5e340a65f01f","display":"tony+1234@medplum.com"},"strictMode":true,"id":"8aee2f1e-6278-44a6-b9b1-f37fac33b14d","meta":{"versionId":"20fc4b22-c1c4-4299-9312-414461def19f","lastUpdated":"2023-04-10T23:42:44.111Z","author":{"reference":"Practitioner/f0fb606e-2c6a-4ba7-a170-cfb6ebe2926a","display":"Tony Lee"},"project":"8aee2f1e-6278-44a6-b9b1-f37fac33b14d","compartment":[{"reference":"Project/8aee2f1e-6278-44a6-b9b1-f37fac33b14d"}]},"description":"Has 1600 sample patients with 8700 Diagnostic Samples"}
{"resourceType":"Project","name":"Massachusetts Sample Data","owner":{"reference":"User/c3c2eb49-0317-438e-b42c-5e340a65f01f","display":"tony+1234@medplum.com"},"strictMode":true,"id":"d2c65e1e-793b-4faa-b3a3-b0b2e8b1f847","meta":{"versionId":"59e0e3d8-bd6e-4222-9b8a-453e389d294a","lastUpdated":"2023-04-10T23:44:41.942Z","author":{"reference":"Practitioner/93ea230d-323c-410f-aeb7-98b6e96b21ab","display":"Medplum Admin"},"project":"d2c65e1e-793b-4faa-b3a3-b0b2e8b1f847","compartment":[{"reference":"Project/d2c65e1e-793b-4faa-b3a3-b0b2e8b1f847"}]}}
{"resourceType":"Project","name":"Super Admin","owner":{"reference":"User/b7ce3c34-7f49-44dc-9721-bcc56c2ba0d3","display":"admin@example.com"},"superAdmin":true,"id":"3a6968bf-91d4-4140-a572-97f979857270","meta":{"versionId":"a4b20d32-59cf-4dd8-9f35-31024ccf2d26","lastUpdated":"2023-04-11T22:26:29.007Z","author":{"reference":"Practitioner/93ea230d-323c-410f-aeb7-98b6e96b21ab","display":"Medplum Admin"},"project":"3a6968bf-91d4-4140-a572-97f979857270","compartment":[{"reference":"Project/3a6968bf-91d4-4140-a572-97f979857270"}]},"features":["bots","cron"]}
```

### CODE REVIEW

File: packages/cli/src/index.ts [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import { MEDPLUM_VERSION, MedplumClient, normalizeErrorString } from '@medplum/core';
 import { Command } from 'commander';
 import dotenv from 'dotenv';
+import fetch from 'node-fetch';
```
Comment (tonylee80) [inline packages/cli/src/index.ts:4]: remove, was used for debugging locally

File: packages/cli/src/bulkExport.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { Command } from 'commander';
+import { writeFile } from 'fs';
+import { medplum } from '.';
+
+export const bulkExport = new Command('bulk-export');
+
```
Comment (jamestouri) [inline packages/cli/src/bulkExport.ts:6]: If someone were to type `medplum bulkExport` what will happen?

**Review by jamestouri — APPROVED** [review]

File: packages/cli/src/bulkExport.ts (hunk 2 above)
Comment (tonylee80) [reply packages/cli/src/bulkExport.ts:6]: Good question! I'll create the cli docs as a follow up task. 

`medplum bulk-export` will perform a system level export which exports all resources as per https://www.medplum.com/docs/api/fhir/operations/bulk-fhir#system-level-export. 
[Fhir System Level Export](https://www.medplum.com/docs/api/fhir/operations/bulk-fhir#system-level-export)

**Review by codyebberson — APPROVED** [review]

**Comment by codyebberson** [comment]
Oh, pulling into the `MedplumClient` PR, perfect, thanks @tonylee80 !

https://github.com/medplum/medplum/pull/2071/files

**Pull request merged**